what4/src/What4/Utils/BVDomain/Arith.hs

```diff
@@ -124,6 +126,11 @@ data Domain (w :: Nat)
   -- first argument caches the value @2^w-1@.
   deriving Show
 
+sameDomain :: Domain w -> Domain w -> Bool
```

Comment:
why not `deriving Eq`?

Reply:
I thought about it.  Datatype equality coincides with "semantic" equality right now, but in principle does't have to, and I sort of wanted to allow for that.